css/BEM-style.md

```diff
@@ -0,0 +1,223 @@
+
+**1.Understanding what BEM is?:** 
+
+BEM is a naming methodology/principle which stand for "Block", "Element", "Modifier" to describe and differentiate, the different components of your webpage in a way that is easy to understand,maintain and scale as and when required.
+In BEM methodology the important point to note is that each and every HTML element will be accessed using a class only, we don't use ID's or tags to style them, to keep specificity uniform across the entire project i.e. (0,1,0) : (ID's, Classes, Tags).
+
+- The **'Block'** are the standalone components in a webpage that are meaningful on its own.
+ 
+  >Example includes: navbar,header,footer,hero-boxes, buttons etc.
+
+-  The **'Element'** are the components that have no standalone meaning, but when tied to a block, represents a part of the block. It expresses a semantic meaning only when tied to a block.  
+   
+   >Example includes: nav-links,footer-links, headings and sub headings inside heroboxes etc.
+
+-  The **'Modifier'** are like flags that differentiate blocks or elements inside block in terms of its appreance.
+   
+   >Example includes: buttons which are muted, navlinks which are active, disabled etc. 
+
+
+**2.Understanding how BEM is written (The Syntax) ?:** 
+
+To understand this, let us consider that we are making two different buttons: one which is in active state and one which is in ghosted state.
+Then the HTML makrup with the classes will be:
+```
+<a href="#"class="btn btn--active">I know!</a>
+<a href="#"class="btn btn--ghosted">I surely know!</a>
+```
+
+Notice the pattern in naming of the classes, ***.btn*** is the block since it has a standalone meaning, while ***.btn--active*** and ***.btn--ghosted*** are modifiers which do not have a standalone meaning but when sematically attached to it's block (the button) describe the modification (flags) performed on the button (i.e) active modifier/flag and ghosted modifier/flag.  
+The modifiers are named by appeneding `--` to the element which you wish to modify.
+
+Now these buttons will be a part of hero-box or any call to action section. We'll  write the markup that describes that, with the necessary classes.
+ 
+```
+<div class="hero-box__buttons">
+    <a href="#"class="btn btn--active">I know!</a>
+    <a href="#"class="btn btn--ghosted">I surely know!</a>
+</div>
+```
+
+Notice a new class naming of ***.hero-box__buttons***. This describes the element called buttons inside a block called hero box with class name of ***.hero-box***. This is how elements inside a block are named, `__` is appended to a element present inside a block ,to describe that this element of buttons are part of hero-box. Note that there can be many groups of buttons inside a website, so these naming styles immediately tell us , it is part of which block.
+
+Now there can also be other elements inside the block hero-box so, we'll write the markup with neccessary classes to describe that:
+```
+<div class="hero-box__text">
```

Comment:
I don't understand the purpose of the `<div>` here. Why is it needed? Why can't `hero-box__text` be applied to `<h1>` directly?

Reply:
This would just be a styling preference because it might so happen there exists another sub-primary-heading inside the text box as well.